core/src/modules/ops/profile.ts

```diff
@@ -469,16 +470,46 @@ export namespace ProfileOps {
   export async function _export(
     profile: Profile,
     force = false,
-    oldGroupsOverride: Group[] = [],
+    oldGroups: Group[] = [],
     saveExports = true
   ) {
     const groups = await profile.$get("groups");
 
     const destinations = await Destination.destinationsForGroups([
+      ...oldGroups,
       ...groups,
-      ...oldGroupsOverride,
     ]);
 
+    // We want to find destinations which aren't in the above set and already have an Export for this Profile.
+    // That's a sign that the Profile is about to get a toDelete export
+    const existingExportNotDeleted: { destinationId: string }[] =
+      await api.sequelize.query(
+        `
+    SELECT * from "exports"
+    JOIN (
+      SELECT "destinationId", MAX("createdAt") as "createdAt"
+      FROM "exports"
+      WHERE "profileId" = '${profile.id}'
+      GROUP BY "destinationId"
+    ) AS "latest"
+    ON "latest"."destinationId" = "exports"."destinationId" AND "latest"."createdAt" = "exports"."createdAt"
+    WHERE "profileId" = '${profile.id}'
+    AND "toDelete" = false
+    ;
```

Comment:
This is pretty inefficient, but it's only for one profile at a time, and only on user-request (not used in tasks)